### DISCUSSION TITLE
Missing many 'standard' settings, compared to vscode [mac]

### DISCUSSION THREAD

**Discussion by marjo8** [eclipse-theia/theia#14558] [General]
Compared to vscode, I miss many settings that I would consider standard or basic settings in Theia. Eamples are ways to specify the light and dark themes, specify certain tokenisation options, options to make Theia use standard tabs and such on my mac, window size inheritance, which shell to use in terminal, and more.
I copied the settings from my vscode settings file in the hope they would work anyway. Alas, this does not seem to be the case.
For the missing settings I get a problem notification (warning) on the settings file that says `<property-name> is not allowed`

I thought that Theia is based on vscode, so I am wondering why these options are not available. The options bring me a lot of value in the IDE.
Will Theia support all those missing options in the (not too distant) future, or would it be better for me to continue using vscode?
I hope the former, as Theia also contains some things I really like that are not available in vscode. And using a fully open and open source IDE is its own reward as well.

**Comment by msujew** [comment]
Hey @marjo8,

please consider creating a actionable issue for this, where we can better coordinate contributor efforts.

> I thought that Theia is based on vscode, so I am wondering why these options are not available.

Theia is not a fork of VS Code, and we share very little code, except for API compatibility. It is a standalone project with distinct goals. Every feature (whether also supported by VS Code or not) is written from scratch.

> Will Theia support all those missing options in the (not too distant) future, or would it be better for me to continue using vscode?

Our roadmap isn't really fine-grained (you will find it [here](https://github.com/eclipse-theia/theia/wiki/Eclipse-Theia-Roadmap)). But we can put up the settings issue in our Theia IDE issue tracker, see https://github.com/eclipse-theia/theia/issues/14119. We attempt to tackle these issues in a combined contributor effort.

**Reply by marjo8** [reply]
Hi @msujew,

That clears up a lot for me. Makes my expectations different.

Shall I open a new issue for the settings things I am missing?
What is the best way to do that? (labels to add, what category of issue)?
Would it be helpful if I include the settings I would hope to see (from the settings.json file) that now are not allowed? There is quite a number of them. Some more important to me than others.
I have reported two issues with settings already. Should every setting I am missing be a separate issue?

**Reply by msujew** [reply]
> Shall I open a new issue for the settings things I am missing?

Yes, that would be really appreciated. You can create them as feature requests. We have an [issue template](https://github.com/eclipse-theia/theia/issues/new?assignees=&labels=&projects=&template=feature_request.md) for that :)

Generally, it usually makes sense to have one issue for each preference. That way, contributors can pick and choose what they want to add to the framework. We're fine with addressing multiple issues in one PR, but it's difficult to have multiple PRs to attempt to close one issue.

As for the missing settings you've mentioned earlier, some of them are already documented. You can take inspiration for your own issues. Labeling will be done by us.

* https://github.com/eclipse-theia/theia/issues/8060
* https://github.com/eclipse-theia/theia/issues/10428

Note that I believe that the framework already supports adjusting tab size (and changing between `\t` and whitespace, if that's what you mean), and which shell to use in the terminal:

https://github.com/eclipse-theia/theia/blob/c8ed4981bef563c4a783e9e2dbcd55587b526748/packages/terminal/src/browser/terminal-preferences.ts#L160-L178

**Reply by marjo8** [reply]
No this doesn't have to do with tab characters. I documented this issue in issue #14385
Another thing I documented in issue  #14386
(how do I put a good link to an issue in a post here?)

I will see when I can find the time on other issues. It takes some time to create the issues for all things I found.

I am trying the terminal profiles thing. Thanks for pointing me in that direction.

**Comment by marjo8** [comment]
And thank you again for the pointer to the terminal settings. This does make terminal use the bash shell:
```   
    "terminal.integrated.defaultProfile.osx": "bash",
    "terminal.integrated.profiles.osx": {
        "zsh": null,
        "bash": {
            "path": "/opt/homebrew/bin/bash"
        }
    },
 ```

**Comment by marjo8** [comment]
and putting the issue number in the message automatically creates a link. Some things are so easy :-)

**Comment by marjo8** [comment]
Hi msujew,

Unfortunately Theia is currently taking too much of my time. In getting it right, I mean. 
The GitKraken GitLens extension does not seem to work. 
The button for the Metals extension keeps disappearing from the sidebar (it's there, but doesn't show its icon). 
I want to use colouring - and am very specific about that. 
Navigating among projects is harder because Theia doen't use Mac window tabs. 
And I don't know what other issues with my extensions (I use many of them) and features I want to see in my IDE will pop up.
 
For me, Theia in the current incarnation does not offer what I need to do my work in as smooth a way as possible. Unfortunately I have to spend more time on the IDE than on my work, when I am using Theia.

To do my work, I will stick with vscode for a while longer. 

I will keep an eye on Theia and try it again in a year or so, repeating that (trying every year), until it gives me what I need and is no longer in the way of doing my work. I really hope to be able to use it with my way of working at some time!

Sorry to have no better news at the moment.

**Reply by msujew** [reply]
All good, these are totally valid complaints. Thanks for trying out Theia and documenting the issues, we appreciate it :)

If open source-ness of your IDE is important to you, I would also recommend trying out [VSCodium](https://vscodium.com/), if you haven't tried it out yet. It's actually a fork of VS Code with all of the Microsoft bits removed. It uses the same [extension store as we do](https://open-vsx.org/), since the official Microsoft store is only accessible by the official VS Code distribution.
